src/secure_tunnel_watcher.py

```diff
@@ -16,6 +16,32 @@
 
 LOCK_FILE_PATH = "/app/lock/"
 
+def extract_http_proxy_url(url):
```

Comment:
How is this different to the Python `urllib.parse(...)` function?https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse